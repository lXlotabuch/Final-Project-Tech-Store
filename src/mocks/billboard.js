/* eslint-disable import/prefer-default-export */
export const billboardMock = {
  title: 'gaming monitors',
  img: 'https://fuzzmusic.ru/wp-content/uploads/2020/08/pesni-iz-igry-kiberpank-2077-1.jpg',
  url: '/catalog?categories=gamingMonitors',
  category: 'gaming monitors'
}