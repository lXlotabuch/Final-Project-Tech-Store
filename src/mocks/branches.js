/* eslint-disable import/prefer-default-export */
export const branchesMock = [
  {
    branchName: 'Branch name',
    branchRef: 'kjhkjh3j23kjhkjhk32jhkj4',
  }
]