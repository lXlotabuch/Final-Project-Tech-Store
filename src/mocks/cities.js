/* eslint-disable import/prefer-default-export */
export const citiesMock = [
  {
    CityName: 'Kyiv',
    Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Dnipro',
    Ref: 'db5c88f0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Kharkiv',
    Ref: 'db5c88e0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Lviv',
    Ref: 'db5c88f5-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Odessa',
    Ref: 'db5c88d0-391c-11dd-90d9-001a92567626'
  },
]

export const cityMock = {
  CityName: 'Kyiv',
  Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
}