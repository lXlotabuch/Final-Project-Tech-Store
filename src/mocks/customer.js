export const mockCustomer = {
  customerNo: '82724063',
  date: '2021-02-19T17:12:50.176+00:00',
  _id: '602ff192ad29b70015add9ab',
  email: 'o.titova.ua@gmail.com',
  isAdmin: false,
  enabled: true,
  firstName: 'someName',
  lastName: 'someLastName',
  login: 'admin',
  password: '$2a$10$i7f6tUO3IT.YsrEGzo/6neTIzoxjhhpWh3Hw3Cme3KLmIAeeyKtFS',
  telephone: '+380731005009',
  gender: 'male',
  ratedProducts: [
    '33823',
    '296354',
    '996598',
    '500227',
    '694186',
    '73109',
    '394414',
    '833228',
    '562546',
    '954601',
    '179158',
    '132036'
  ],
}
export default mockCustomer
