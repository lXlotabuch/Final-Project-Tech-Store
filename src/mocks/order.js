/* eslint-disable import/prefer-default-export */
export const orderMock = {
  deliveryAddress: {country: 'Ukraine', city: 'db5c88f0-391c-11dd-90d9-001a92567626', branch: 'Отделение №2: ул. Академика Янгеля, 40'},
  email: 'o.titova.ua@gmail.com',
  firstName: 'test',
  letterHtml: '<h1>Your order is placed. Our manager will contact you soon.</h1>',
  letterSubject: 'test, thank you for order!',
  mobile: '+380660788828',
  orderNo: '8395086',
  paymentInfo: 'Payment at the time of receipt of the goods',
  shipping: 121,
  totalSum: 117400,
  date: '2021-04-13T14:14:27.478Z',
  _id: '6070d130058f4e00154bbe23'
}