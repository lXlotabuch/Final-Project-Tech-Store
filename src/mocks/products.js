export const productMock = {
  _id: '604ffa554e7ef500153393f9',
  name: 'Monitor 27" AOC C27G2ZU/BK',
  enabled: true,
  currentPrice: 10600,
  previousPrice: 12600,
  categories: 'gamingMonitors',
  imageUrls: [
    'https://i8.rozetka.ua/goods/18842039/aoc_c27g2zu_bk_images_18842039923.jpg',
    'https://i2.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040211.jpg',
    'https://i8.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040937.jpg',
    'https://i8.rozetka.ua/goods/18842041/aoc_c27g2zu_bk_images_18842041729.jpg'
  ],
  quantity: 44,
  color: 'black',
  productUrl: '/gamingMonitors',
  brand: 'AOC',
  newProduct: 'no',
  reviews: [4, 2, 3, 5, 4, 5, 3, 4, 3, 5, 5, 4, 5],
  params: {
    updateFrequency: '240Hz',
    countryOfmanufacture: 'China',
    displayDiagonal: '27"',
    displayResolution: '1920x1080',
    responseTime: '0,5ms',
    screenSize: '32 inches',
    aspectRatio: '16:9'
  },
  description: 'An immersive 27" super curved gaming monitor with a fast 144Hz refresh rate and 1ms response time, helping you get the most out of your monitor. Features such as AMD FreeSync Premium, QHD resolution, and a curved design make this monitor ultra competitive for any gamer.',
  itemNo: '554194',
}
const productMock2 = {
  ...productMock,
  itemNo: '554195',
  _id: '604ffa554e7ef500153393f0',
}
export const productsMock = [
  {
    _id: '607057d9e448c200152a5047',
    cartQuantity: 1,
    product: productMock
  },
  {
    _id: '607057d9e448c200152a5048',
    cartQuantity: 1,
    product: productMock2
  }
];