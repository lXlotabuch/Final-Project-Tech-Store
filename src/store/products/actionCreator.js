import {
  ADD_PRODUCT,
  SET_PRODUCTS,
  UPDATE_PRODUCT,
  GET_NEW_PRODUCTS,
  START_LOADING,
  STOP_LOADING,
} from './actionType';

export const setProducts = (payload) => ({
  type: SET_PRODUCTS,
  payload
})

export const addProduct = (payload) => ({
  type: ADD_PRODUCT,
  payload
})

export const updateProduct = (payload) => ({
  type: UPDATE_PRODUCT,
  payload
})
export const getNewProductsCreator = (payload) => ({
  type: GET_NEW_PRODUCTS,
  payload
})
export const startLoading = () => ({type: START_LOADING})
export const stopLoading = () => ({type: STOP_LOADING})