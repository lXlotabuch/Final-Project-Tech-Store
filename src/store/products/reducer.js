import {
  SET_PRODUCTS,
  ADD_PRODUCT,
  UPDATE_PRODUCT,
  GET_NEW_PRODUCTS,
  START_LOADING,
  STOP_LOADING,
} from './actionType';
import shuffleArray from '../../utils/shuffleArray'

export const MODULE_NAME = 'products'
export const selectNewProducts = (state) => state[MODULE_NAME].newProducts
export const selectIsLoading = (state) => state[MODULE_NAME].isLoading

const initialState = {
  products: [],
  newProducts: [],
  pageProduct: {},
  isLoading: false
}

export const reducer = (state = initialState, {type, payload}) => {
  switch (type) {
    case SET_PRODUCTS:
      return {
        ...state,
        products: payload
      }

    case ADD_PRODUCT:
      return {
        ...state,
        products: [payload, ...state.products]
      }

    case UPDATE_PRODUCT:
      return {
        ...state,
        products: state.products.map((el) => (+el.itemNo !== +payload.itemNo ? el : payload))
      }
      
    case GET_NEW_PRODUCTS:
      return {
        ...state,
        newProducts: shuffleArray(payload)
      }

    case START_LOADING:
      return {
        ...state,
        isLoading: true
      }

    case STOP_LOADING:
      return {
        ...state,
        isLoading: false
      }
      
    default:
      return state
  }
}

export default reducer