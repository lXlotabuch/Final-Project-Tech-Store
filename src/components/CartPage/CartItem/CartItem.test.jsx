/* eslint-disable no-underscore-dangle */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history'
import {
  fireEvent, render, screen, waitFor
} from '@testing-library/react'
import { machMedia } from '../../../mocks/matchMedia.mock';
import {CartItemComponent} from './CartItem'
import { productMock } from '../../../mocks/products';

describe('CartItemComponent', () => {
  beforeEach(() => {
    machMedia()
  })
  test('render', () => {
    const history = createMemoryHistory()
    const {asFragment} = render(
      <Router history={history}>
        <CartItemComponent
          product={productMock}
          cartQuantity={1}
          increaseQuantity={() => {}}
          decreaseQuantity={() => {}}
          removeFromCart={() => {}}
        />
      </Router>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('handle decrease quantity', async () => {
    const decreaseQuantity = jest.fn()
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <CartItemComponent
          product={productMock}
          cartQuantity={1}
          increaseQuantity={() => {}}
          decreaseQuantity={decreaseQuantity}
          removeFromCart={() => {}}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /minus/i}))
    await waitFor(() => expect(decreaseQuantity).toBeCalled())
    expect(decreaseQuantity.mock.calls[0]).toEqual([productMock._id])
  })

  test('handle increase quantity', async () => {
    const increaseQuantity = jest.fn()
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <CartItemComponent
          product={productMock}
          cartQuantity={1}
          increaseQuantity={increaseQuantity}
          decreaseQuantity={() => {}}
          removeFromCart={() => {}}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /plus/i}))
    await waitFor(() => expect(increaseQuantity).toBeCalled())
    expect(increaseQuantity.mock.calls[0]).toEqual([productMock])
  })
  
  test('handle remove', async () => {
    const removeFromCart = jest.fn()
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <CartItemComponent
          product={productMock}
          cartQuantity={1}
          increaseQuantity={() => {}}
          decreaseQuantity={() => {}}
          removeFromCart={removeFromCart}
        />
      </Router>
    )
    fireEvent.click(screen.getByRole('button', {name: /close/i}))
    await waitFor(() => expect(removeFromCart).toBeCalled())
    expect(removeFromCart.mock.calls[0]).toEqual([productMock._id])
  })
})