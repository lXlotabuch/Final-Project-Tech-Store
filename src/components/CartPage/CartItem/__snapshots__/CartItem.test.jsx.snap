// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItemComponent render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ btQEoA"
  >
    <div
      class="ant-row ant-row-middle"
    >
      <div
        class="ant-col ant-col-xs-8 ant-col-md-8 ant-col-lg-5"
      >
        <a
          href="/products/554194"
        >
          <img
            alt="Monitor 27\\" AOC C27G2ZU/BK"
            src="https://i8.rozetka.ua/goods/18842039/aoc_c27g2zu_bk_images_18842039923.jpg"
          />
        </a>
      </div>
      <div
        class="ant-col ant-col-xs-16 ant-col-md-16 ant-col-lg-7"
      >
        <a
          href="/products/554194"
        >
          <p
            class="bold"
          >
            Monitor 27" AOC C27G2ZU/BK
          </p>
        </a>
      </div>
      <div
        class="ant-col ant-col-xs-5 ant-col-md-7 ant-col-lg-3"
      >
        <span
          class="price"
        >
          10600₴
        </span>
      </div>
      <div
        class="ant-col ant-col-xs-7 ant-col-md-8 ant-col-lg-5"
      >
        <div
          class="sc-hKgILt sc-eCssSg bGMZSh eZYWkv"
        >
          <button
            class="ant-btn ant-btn-circle ant-btn-icon-only"
            type="button"
          >
            <span
              aria-label="minus"
              class="anticon anticon-minus"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="minus"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
          </button>
          <input
            class="ant-input sc-dlfnbm hioKae"
            type="text"
            value="1"
          />
          <button
            class="ant-btn ant-btn-circle ant-btn-icon-only"
            type="button"
          >
            <span
              aria-label="plus"
              class="anticon anticon-plus"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="plus"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                />
                <path
                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-col subtotal ant-col-xs-9 ant-col-md-6 ant-col-lg-3"
      >
        <span>
          10600₴
        </span>
      </div>
      <div
        class="ant-col ant-col-xs-2 ant-col-md-3 ant-col-lg-1"
      >
        <button
          class="ant-btn ant-btn-circle ant-btn-icon-only sc-gsTCUz ftEqYG"
          type="button"
        >
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
